File: Easy/238.go
package easy

func productExceptSelf(nums []int) []int {
	left := make([]int, len(nums))
	left[0] = 1
	right := make([]int, len(nums))
	right[len(nums)-1] = 1

	for i := 2; i < len(nums); i++ {
		left[i] = left[i-1] * nums[i-1]
	}

	for i := len(nums) - 2; i >= 0; i-- {
		right[i] = right[i+1] * nums[i+1]
	}

	for i := 0; i < len(nums); i++ {
		nums[i] = left[i] * right[i]
	}
	return nums
}

File: Easy/238.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans = vector<int>(nums.size(), 1);
        ans[0] = 1;

        for(int i = 1; i < nums.size(); i++) {
            ans[i] = ans[i - 1] * nums[i - 1];
        }

        int right = 1;

        for( int i = nums.size() - 2; i >=0; i--) {
            right *= nums[i + 1];
            ans[i] *= right;
        }

        return ans;
    }
};